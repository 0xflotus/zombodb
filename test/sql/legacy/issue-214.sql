CREATE TABLE issue_214 (id SERIAL PRIMARY KEY, name varchar);
CREATE INDEX idx_zdb_foo ON issue_214 USING zombodb((issue_214.*));
INSERT INTO issue_214 (name) VALUES ('a!a');

SELECT * FROM zdb.dump_query('idx_zdb_foo', '_all:a\!*');
SELECT assert(zdb.dump_query('idx_zdb_foo', '_all:a\!*'), zdb.dump_query('idx_zdb_foo', '_all:"a!*"'), '_all: wildcard escaped v/s not');
SELECT * FROM issue_214 WHERE issue_214 ==> '_all:a\!*';


SELECT * FROM zdb.dump_query('idx_zdb_foo', 'name:a\!*');
SELECT assert(zdb.dump_query('idx_zdb_foo', 'name:a\!*'), zdb.dump_query('idx_zdb_foo', 'name:"a!*"'), 'name: wildcard escaped v/s not');
SELECT * FROM issue_214 WHERE issue_214 ==> 'name:a\!*';

DROP TABLE issue_214;
