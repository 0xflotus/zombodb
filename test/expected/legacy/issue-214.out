CREATE TABLE issue_214 (id SERIAL PRIMARY KEY, name varchar);
CREATE INDEX idx_zdb_foo ON issue_214 USING zombodb((issue_214.*));
INSERT INTO issue_214 (name) VALUES ('a!a');
SELECT * FROM zdb.dump_query('idx_zdb_foo', '_all:a\!*');
             dump_query             
------------------------------------
 {                                 +
   "prefix": {                     +
     "zdb_all": {                  +
       "boost": 1.0,               +
       "rewrite": "constant_score",+
       "value": "a!"               +
     }                             +
   }                               +
 }
(1 row)

SELECT assert(zdb.dump_query('idx_zdb_foo', '_all:a\!*'), zdb.dump_query('idx_zdb_foo', '_all:"a!*"'), '_all: wildcard escaped v/s not');
 assert 
--------
 t
(1 row)

SELECT * FROM issue_214 WHERE issue_214 ==> '_all:a\!*';
 id | name 
----+------
(0 rows)

SELECT * FROM zdb.dump_query('idx_zdb_foo', 'name:a\!*');
             dump_query             
------------------------------------
 {                                 +
   "prefix": {                     +
     "name": {                     +
       "boost": 1.0,               +
       "rewrite": "constant_score",+
       "value": "a!"               +
     }                             +
   }                               +
 }
(1 row)

SELECT assert(zdb.dump_query('idx_zdb_foo', 'name:a\!*'), zdb.dump_query('idx_zdb_foo', 'name:"a!*"'), 'name: wildcard escaped v/s not');
 assert 
--------
 t
(1 row)

SELECT * FROM issue_214 WHERE issue_214 ==> 'name:a\!*';
 id | name 
----+------
  1 | a!a
(1 row)

DROP TABLE issue_214;
